src/gui/qgsqueryresultwidget.cpp

```diff
@@ -732,6 +745,77 @@ void QgsQueryResultWidget::updateDialogTitle()
   emit requestDialogTitleUpdate( fileName );
 }
 
+void QgsQueryResultWidget::populatePresetQueryMenu()
+{
+  mPresetQueryMenu->clear();
+
+  QMenu *storeQueryMenu = new QMenu( tr( "Store Current Query" ), mPresetQueryMenu );
+  mPresetQueryMenu->addMenu( storeQueryMenu );
+  QAction *storeInProfileAction = new QAction( tr( "In User Profile…" ), storeQueryMenu );
+  storeQueryMenu->addAction( storeInProfileAction );
+  storeInProfileAction->setEnabled( !mSqlEditor->text().isEmpty() );
+  connect( storeInProfileAction, &QAction::triggered, this, [this] {
+    storeCurrentQuery( Qgis::QueryStorageBackend::LocalProfile );
+  } );
+  QAction *storeInProjectAction = new QAction( tr( "In Current Project…" ), storeQueryMenu );
+  storeQueryMenu->addAction( storeInProjectAction );
+  storeInProjectAction->setEnabled( !mSqlEditor->text().isEmpty() );
+  connect( storeInProjectAction, &QAction::triggered, this, [this] {
+    storeCurrentQuery( Qgis::QueryStorageBackend::CurrentProject );
+  } );
+
+
+  const QList< QgsStoredQueryManager::QueryDetails > storedQueries = QgsGui::storedQueryManager()->allQueries();
```

Comment:
Just a note, this may be not very convenient when user has A LOT of saved queries. But I do not know how common this scenario is and, unfortunately, right now I do not have a better idea how to implement it.

Reply:
Yeah, I realise there's an inherent limit in the number of queries the UI will handle. I couldn't think of an alternative though, and the current db manager UI has had a similar constraint for years now without any requests to improve.